### PR TITLE
Better error handling when uploading images via SOAP API

### PR DESCRIPTION
**Pull request by woutersamaey**
### Description (*)
If Magento deems the image invalid, it won't be written to the disk. Yet the code continues like it was written successfully. This results in an inaccurate error message.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by fballiano — APPROVED** [review]

**Review by ADDISON74 — DISMISSED** [review]

**Review by fballiano — APPROVED** [review]

**Review by luigifab — COMMENTED** [review]
Is someone have some lines of PHP to test it?

**Comment by fballiano** [comment]
I've fixed the conflict, @woutersamaey would you please rebase it to "main"? I can't do it unfortunately

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php [hunk 1]
```diff
@@ -144,12 +144,18 @@ public function create($productId, $data, $store = null, $identifierType = null)
...
 
             // try to create Image object - it fails with Exception if image is not supported
             try {
+                // Write image file
+                $fileWritten = $ioAdapter->write($fileName, $fileContent, 0666);
+                if ($fileWritten === false) {
+                    throw new Mage_Core_Exception('Could not write image file to disk. Invalid image data?');
+                }
+                unset($fileContent);
+
```
Comment (fballiano) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php:158]:
```suggestion
            
```

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php [hunk 2]
```diff
@@ -144,12 +144,18 @@ public function create($productId, $data, $store = null, $identifierType = null)
...
-            // Write image file
             $ioAdapter->write($fileName, $fileContent, 0666);
             unset($fileContent);
 
             // try to create Image object - it fails with Exception if image is not supported
             try {
+                // Write image file
+                $fileWritten = $ioAdapter->write($fileName, $fileContent, 0666);
+                if ($fileWritten === false) {
+                    throw new Mage_Core_Exception('Could not write image file to disk. Invalid image data.');
```
Comment (fballiano) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php:155]:
```suggestion
                    throw new Mage_Core_Exception('Could not write image file to disk. Invalid image data?');
```

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php (hunk 1 above)
Comment (fballiano) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php:158]:
```suggestion

```

**Comment by fballiano** [comment]
Also, PHPStan is not happy.

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php [hunk 3]
```diff
@@ -180,8 +185,10 @@ public function create($productId, $data, $store = null, $identifierType = null)
 
             $product->save();
         } catch (Mage_Core_Exception $e) {
+            Mage::logException($e);
             $this->_fault('not_created', $e->getMessage());
         } catch (Exception $e) {
+            Mage::logException($e);
             $this->_fault('not_created', Mage::helper('catalog')->__('Cannot create image.'));
```
Comment (fballiano) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Media/Api.php:192]:
```suggestion
            $this->_fault('not_created', $e->getMessage());
```

**Comment by fballiano** [comment]
Screenshot 2024-02-14 alle 14 19 56
this image shows that the added code seems to be wrong because what is trying to write is coming from a variable that is unset previously.

So, for the time being, I'll close this PR.

**Pull request closed without merging**